### PR TITLE
Port to Qt 6, stay compatible with at least Qt 5.15

### PR DESCRIPTION
**Pull request by krf**
Note: Tried to make it compatible with Qt 5.12 again, but then ran into issues like missing `QDate::startOfDay` (cf. https://doc.qt.io/qt-5/qdate.html) and then lost motivation.

Could be fixed with some more ifdefs, but I would first like to check "how low" we want to go, i.e. whether Qt 5.14 as minimum required version would be sufficient for KDChart master or not.

Update:
- 5.15 is now the minimum required version
- Tests: Built the following configs: Qt6-dev+CMake, Qt6-dev+QMake, Qt5.15.3+CMake

### CODE REVIEW

File: src/KDChart/KDChartWidget.h [hunk 1]
```diff
@@ -26,15 +26,11 @@
 #include "KDChartGlobal.h"
 
 #include <QWidget>
+#include <QtContainerFwd>
```
Comment (nicolasfella) [inline src/KDChart/KDChartWidget.h:29]: IIRC there is an issue where QtContainerFwd is missing something in 5.14 but 5.15 works fine. Not sure if applicable here though

**Comment by ivan-cukic** [comment]
There's already a Qt 6 branch (`qt6-dev`), with removed warnings, tested and generally stable. Though that one does not support Qt 5 anymore. I'd rather if you made that one Qt5 compatible instead of doing this.

File: src/KDGantt/kdganttview.cpp [hunk 2]
```diff
@@ -611,9 +611,10 @@ void View::print( QPainter* painter, qreal start, qreal end, const QRectF& targe
               drawColumnLabels);
 }
 
-
 #include "moc_kdganttview.cpp"
 
+#undef d
```
Comment (nicolasfella) [inline src/KDGantt/kdganttview.cpp:616]: oof. I think this warrants a comment why

Comment (krf) [reply src/KDGantt/kdganttview.cpp:616]: Actually it's done in a couple .cpp files already -- I think it's clear.

File: src/KDChart/KDChartWidget.h (hunk 1 above)
Comment (krf) [reply src/KDChart/KDChartWidget.h:29]: Honestly I wouldn't mind just using Qt 5.15 as minimum required version, too. I'll wait if someone wants to keep this compatible with a lower Qt version at all.

**Comment by krf** [comment]
> There's already a Qt 6 branch (`qt6-dev`), with removed warnings, tested and generally stable. Though that one does not support Qt 5 anymore. I'd rather if you made that one Qt5 compatible instead of doing this. At least, that was my plan the next time I get some KDChart hours.

I have to admit I looked only briefly at your qt6-dev branch. it looked like an experiment to me (lots of unrelated changes, like those `[[nodiscard]]` additions, code reformatting changes, a modernize-use-auto run -- and all that in a bulk commit. Thought it was just an early attempt at it.

Maybe rather use my branch as base (uses easy to revert standalone commits for non-trivial patches), and apply yours on top? Most of your changes will just apply cleanly as it's been done in my branch already.

**Comment by ivan-cukic** [comment]
I ended up squishing all the commits for some reason (don't remember what the reason was).

As for the merge, ok, but don't do it into master just yet. This needs to be communicated with the release team. @winterz had plans for which release should be the 'drop old 5.x' release.

**Comment by krf** [comment]
How shall we deal with this MR?

**Comment by dfaure-kdab** [comment]
I'm mostly out of the loop about KDChart plans, but +1 on the approach, it's exactly what I did for KDSoap and KDReports: supporting both Qt5 and Qt6 in master. I'm not sure why we should "drop 5.x" if we can easily support both for a while.

@winterz any input?

**Comment by ivan-cukic** [comment]
The drop 5.x was meant for old 5.x, not Qt 5 altogether.

I'm fine with this getting merged.

**Comment by dfaure-kdab** [comment]
Oh I see, sorry for the misunderstanding.

File: src/KDChart/Cartesian/KDChartLeveyJenningsDiagram.cpp [hunk 3]
```diff
@@ -531,7 +531,7 @@ void LeveyJenningsDiagram::paint( PaintContext* ctx )
     const QAbstractItemModel& m = *model();
     const int rowCount = m.rowCount( rootIndex() );
 
-    const unsigned int minTime = timeRange().first.toTime_t();
+    const unsigned int minTime = timeRange().first.toSecsSinceEpoch();
```
Comment (dfaure-kdab) [inline src/KDChart/Cartesian/KDChartLeveyJenningsDiagram.cpp:534]: Doesn't this have a year-2106 bug? The whole point of toSecsSinceEpoch() is to store the result in qint64.

File: src/KDChart/KDChartDatasetProxyModel.cpp [hunk 4]
```diff
@@ -51,7 +51,7 @@ void DatasetProxyModel::setDatasetRowDescriptionVector(
                 "A source model must be set before the selection can be configured." );
     initializeDatasetDecriptors( configuration, sourceModel()->rowCount(mRootIndex),
                                  mRowSrcToProxyMap,  mRowProxyToSrcMap );
-    clear(); // clear emits layoutChanged()
+    invalidate(); // clear emits layoutChanged()
```
Comment (dfaure-kdab) [inline src/KDChart/KDChartDatasetProxyModel.cpp:54]: The comment is now outdated :)

File: src/KDChart/KDChartLayoutItems.cpp [hunk 5]
```diff
@@ -187,7 +187,7 @@ void KDChart::TextBubbleLayoutItem::paint( QPainter* painter )
     const QBrush oldBrush = painter->brush();
     painter->setPen( Qt::black );
     painter->setBrush( QColor( 255, 255, 220 ) );
-    painter->drawRoundRect( geometry(), 10 );
+    painter->drawRoundedRect( geometry(), 10, 25 ); // KDAB_QT6_PORT: Double check if '25' is correct
```
Comment (dfaure-kdab) [inline src/KDChart/KDChartLayoutItems.cpp:190]: The deprecation note says you should pass Qt::RelativeSize as an additional argument.

File: src/KDChart/Cartesian/KDChartLeveyJenningsDiagram.cpp (hunk 3 above)
Comment (krf) [reply src/KDChart/Cartesian/KDChartLeveyJenningsDiagram.cpp:534]: This and below comments addressed in https://github.com/KDAB/KDChart/pull/23

**Pull request merged**